app/forms/candidate_interface/degree_wizard.rb

```diff
@@ -0,0 +1,447 @@
+      Hesa::Institution.find_by_name(institution_name)&.hesa_code
+    end
+
+    def hesa_type_code(type_description)
+      Hesa::DegreeType.find_by_name(type_description)&.hesa_code
+    end
+
+    def hesa_subject_code(subject)
+      Hesa::Subject.find_by_name(subject)&.hesa_code
+    end
+
+    def hesa_grade_code(grade)
+      Hesa::Grade.find_by_description(grade)&.hesa_code
+    end
+
+    def qualification_type_attributes
+      return 'Level 6 Diploma' if degree_level == 'Level 6 Diploma'
+
+      equivalent_level || other_type || type
+    end
+
+    def grade_attributes
+      return other_grade if grade == 'Other'
+
+      grade || other_grade
+    end
+
+    def predicted_grade
+      completed == 'No'
+    end
+
+    def completed?
+      completed == 'Yes'
+    end
+
+    def uk?
+      uk_or_non_uk == 'uk'
+    end
+
+    def other_qualification?
+      degree_level == 'Another qualification equivalent to a degree'
+    end
+
+    def level_options?
+      ['Level 6 Diploma', 'Another qualification equivalent to a degree'].include?(degree_level)
+    end
+
+    def other_type_selected
+      type == "Another #{degree_level.split.first.downcase} degree type"
+    end
+
+    def other_type_filled_out
+      other_type.present?
+    end
+
+    def grade_choices
+      grade == 'Other' || grade == 'Yes'
+    end
+
+    def subjects
+      @subjects ||= Hesa::Subject.names.sort
+    end
+
+    def institutions
+      @institutions ||= Hesa::Institution.names.sort
+    end
+
+    def dynamic_type(degree_level)
+      return if degree_level.nil?
+      return 'doctorate' if degree_level == 'Doctorate (PhD)'
+
+      degree_level.to_s.to_s.downcase
+    end
+
+  private
+
+    def award_year_is_before_start_year
+      errors.add(:award_year, :before_the_start_year) if start_year.present? && award_year.to_i < start_year.to_i
+    end
+
+    def start_year_is_after_award_year
+      errors.add(:start_year, :after_the_award_year) if award_year.present? && start_year.to_i > award_year.to_i
+    end
+
+    def start_year_in_future_when_degree_completed
+      errors.add(:start_year, :in_the_future) if completed? && start_year.present? && start_year.to_i >= RecruitmentCycle.next_year
+    end
+
+    def award_year_in_future_when_degree_completed
+      errors.add(:award_year, :in_the_future) if completed? && award_year.present? && award_year.to_i >= RecruitmentCycle.next_year
+    end
+
+    def award_year_in_past_when_degree_incomplete
+      errors.add(:award_year, :in_the_past) if start_year.present? && predicted_grade && award_year.to_i < recruitment_cycle_year.to_i
+    end
+
+    def award_year_after_teacher_training_starts
+      errors.add(:award_year, :after_teacher_training) if predicted_grade && award_year.to_i > recruitment_cycle_year.to_i
+    end
+
+    def map_value_for_no_submitted_international_grade(grade)
+      self.grade = {
+        'No' => 'N/A',
+        'I do not know' => 'Unknown',
```

Comment:
Something for another PR but we really shouldn't be storing magic values like 'N/A' or 'Unknown' in the `grade` attribute. There should be a separate attribute for these kinds of things.

Reply:
For would it be useful 'N/A' and 'Unknown' into constants?

Reply:
I've made these string values constants